packages/actor-rdf-parse-html-rdfa/lib/ActorRdfParseHtmlRdfa.ts

```diff
@@ -0,0 +1,34 @@
+import {ActorRdfParseHtml, IActionRdfParseHtml, IActorRdfParseHtmlOutput} from "@comunica/bus-rdf-parse-html";
+import {IActorArgs, IActorTest} from "@comunica/core";
+import {RdfaParser} from "rdfa-streaming-parser";
+
+/**
+ * A comunica RDFa RDF Parse Html Actor.
+ */
+export class ActorRdfParseHtmlRdfa extends ActorRdfParseHtml {
+
+  constructor(args: IActorArgs<IActionRdfParseHtml, IActorTest, IActorRdfParseHtmlOutput>) {
+    super(args);
+  }
+
+  public async test(action: IActionRdfParseHtml): Promise<IActorTest> {
+    return true;
+  }
+
+  public async run(action: IActionRdfParseHtml): Promise<IActorRdfParseHtmlOutput> {
+    const mediaType = action.headers ? action.headers.get('content-type') : null;
+    const language = action.headers ? action.headers.get('content-language') : null;
+    const profile = mediaType && mediaType.indexOf('xml') >= 0 ? 'xhtml' : 'html';
+
+    const htmlParseListener = new RdfaParser({ baseIRI: action.baseIRI, profile, language });
+    htmlParseListener.on('error', action.error);
+    htmlParseListener.on('data', action.emit);
+    const onTagEndOld = htmlParseListener.onEnd;
```

Comment:
This is needed to make sure that `action.end()` is called _after_ the RFDa parser has ended.

Reply:
But why is it like this and not `htmlParseListener.on('end', () => action.end() );`. Is there some hidden postprocessing done after firing the end event?

Reply:
Ah yes. That is because the RdfaParser itself does quite a bit of work when that end event is received. By doing it like this, we ensure that `action.end()` is only called _after_ the RdfaParser is completely finished, so that we don't loose any triples that are emitted at the end.